### ISSUE TITLE
Abide behavior on a visually hidden input element

### ISSUE THREAD

**Issue by gormus** [zurb/foundation-sites#10972]
Omitting `data-abide-ignore` attribute from an `input type="text" style="display: none"` element unexpectedly hides all the error messages marked with `.form-error` classes in the form. However elements with validation errors are still rendered with red colors and borders. Note that the element doesn't have the `required` attribute.

#### How to reproduce this bug:

1. Submit the form and observe the red-highlighted elements without errot messages.
2. Add `data-abide-ignore` attribute to the `input type="text" name="_gotcha" style="display:none"` element, and submit the form again. Now error messages are shown along with the red colors and border as expected.

#### What should happen:
I expected to see only the elements with `required` attribute validated with Abide. 

#### What happened instead:
Having `input type="text" style="display: none"` element partly broke Abide's behavior. All elements with `required` attribute are rendered with red colors and borders but without error messages.

#### Test case link:
https://codepen.io/gormus/pen/OQEMrY

#### Before submitting the form:
<img width="806" alt="screen shot 2018-02-21 at 6 03 05 pm" src="https://user-images.githubusercontent.com/291025/36518066-c911c50c-1739-11e8-82ab-f17304b430e1.png">

#### Submitted without `data-abide-ignore` attribute:
<img width="806" alt="screen shot 2018-02-21 at 6 03 23 pm" src="https://user-images.githubusercontent.com/291025/36518122-fbe2a852-1739-11e8-882c-18cfbee78722.png">

#### Submitted with added `data-abide-ignore` attribute (expected result):
<img width="806" alt="screen shot 2018-02-21 at 6 04 11 pm" src="https://user-images.githubusercontent.com/291025/36518137-0e8b989c-173a-11e8-9b68-d4bca90e67a0.png">

**Comment by DanielRuf** [comment]
In general `display: none` does not make much sense. Normally hidden form-fields are done with `type="hidden".

I think we need some `:visible` selectors.

**Comment by DanielRuf** [comment]
Probably just some small logic issue in the component.

**Comment by gormus** [comment]
I agree, `display: none` doesn't make much sense as a general rule. Perhaps most common use for it to use as a honeypot for bots, and that's how I am using in that form.

However, seems like I was wrong with my assumption, `display: none` is not the issue. After removing `display: none` the result still is the same. https://codepen.io/gormus/pen/gvKZbv
